src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -46,6 +46,7 @@ final class KeyMatcher {
     private static final int SKIP_KEY_LOOKUP = -1;
     private final JsonMaskingConfig maskingConfig;
     private final TrieNode root;
```

Comment:
Now the KeyMatcher has a `TrieNode` and a `StatefulRadixTrieNode` which is quite confusing. Looking at it for a bit I think one is for JsonPath support and the other isn't? In any case this is something that is not clearly documented (or I couldn't find it) and quite counterintuitive without docs. Maybe we can also rename `TrieNode` to `RadixTrieNode` because the difference between the two is only that one has (JsonPath??) state and the other doesn't? Both seem to be Radix Trie nodes